### ISSUE TITLE
"Debugger exited with status 1" for request=launch even though I'm able to execute the same command myself and then debug with request=attach

### ISSUE THREAD

**Issue by Nowaker** [Shopify/ruby-lsp#1767]
### Operating System

Linux

### Ruby version

3.3.0 / 3.1.4

### Project has a bundle

- [X] Has bundle

### Ruby version manager being used

rvm

### Description

```json
    {
      "name": "Rails",
      "type": "ruby_lsp",
      "request": "launch",
      "program": "puma",
      "env": {
        "RAILS_DEBUG": "true"
      },
    }
```

> Debugger exited with status 1. Check the output channel for more information.

![image](https://github.com/Shopify/vscode-ruby-lsp/assets/611746/0074da70-4af2-4028-a05a-99b8055de41c)

```
2024-01-22 20:32:40.983 [info] (worldmodern) Checking if asdf is available on the path with command: /usr/bin/zsh -i -c 'asdf --version'
2024-01-22 20:32:41.353 [info] (worldmodern) Checking if chruby is available on the path with command: /usr/bin/zsh -i -c 'chruby --version'
2024-01-22 20:32:41.513 [info] (worldmodern) Checking if rbenv is available on the path with command: /usr/bin/zsh -i -c 'rbenv --version'
2024-01-22 20:32:41.673 [info] (worldmodern) Checking if rvm is available on the path with command: /usr/bin/zsh -i -c 'rvm --version'
2024-01-22 20:32:41.885 [info] (worldmodern) Discovered version manager rvm
2024-01-22 20:32:41.885 [info] (worldmodern) Trying to activate Ruby environment with command: /usr/bin/zsh -i -c 'rvm-auto-ruby -rjson -e "STDERR.printf(%{RUBY_ENV_ACTIVATE%sRUBY_ENV_ACTIVATE}, JSON.dump({ env: ENV.to_h, ruby_version: RUBY_VERSION, yjit: defined?(RubyVM::YJIT) }))"' inside directory: /home/nowaker/projekty/modern/worldmodern
2024-01-22 20:32:42.978 [info] (worldmodern) Starting Ruby LSP v0.13.1...

2024-01-22 20:32:43.150 [info] (worldmodern) Rails server is not running. To get Rails features in the editor, boot the Rails server
Ruby LSP is ready

2024-01-22 20:33:18.537 [info] Spawning debugger in directory /home/nowaker/projekty/modern/worldmodern
2024-01-22 20:33:18.537 [info]    Command bundle exec rdbg --open --command --sock-path=/tmp/ruby-lsp-debug-sockets/ruby-debug-worldmodern-0.sock -- puma
2024-01-22 20:33:18.537 [info]    Environment {"RAILS_DEBUG":"true","SHELL":"/usr/bin/zsh","SESSION_MANAGER":"local/nwkr-desktop:@/tmp/.ICE-unix/1681,unix/nwkr-desktop:/tmp/.ICE-unix/1681","WINDOWID":"109051917","COLORTERM":"truecolor","VIRSH_DEFAULT_CONNECT_URI":"qemu:///system","rvm_delete_flag":"0","LDRAWDIR":"/usr/share/ldraw","GNOME_DESKTOP_SESSION_ID":"this-is-deprecated","APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL":"true","rvm_prefix":"/home/nowaker","LESS_TERMCAP_se":"\u001b[0m","LESS_TERMCAP_so":"\u001b[01;44;33m","LC_ADDRESS":"en_US.UTF-8","LC_NAME":"en_US.UTF-8","SSH_AUTH_SOCK":"/run/user/1000/keyring/ssh","CLOJURE_HOME":"/usr/share/clojure","SHELL_SESSION_ID":"a7310ab4de664398a9776b43c9fabdac","ELECTRON_RUN_AS_NODE":"1","MY_RUBY_HOME":"/home/nowaker/.rvm/rubies/ruby-3.3.0","LC_MONETARY":"en_US.UTF-8","NO_AT_BRIDGE":"1","VSCODE_AMD_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","CLOUDSDK_PYTHON_ARGS":"-S","EDITOR":"vim","RUBY_VERSION":"ruby-3.3.0","XDG_SEAT":"seat0","PWD":"/home/nowaker/projekty/modern/worldmodern","LOGNAME":"nowaker","QT_QPA_PLATFORMTHEME":"qt5ct","XDG_SESSION_TYPE":"x11","rvm_version":"1.29.12-next (master)","RADV_PERFTEST":"aco","VSCODE_CODE_CACHE_PATH":"/home/nowaker/.config/Code/CachedData/8b3775030ed1a69b13e4f4c628c612102e30a681","XAUTHORITY":"/home/nowaker/.Xauthority","TF_PLUGIN_CACHE_DIR":"/home/nowaker/.terraform.d/plugin-cache","BUNDLER_ARGS":"-j30","MOTD_SHOWN":"pam","HOME":"/home/nowaker","LANG":"en_US.UTF-8","LC_PAPER":"en_US.UTF-8","LS_COLORS":"rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=00:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.avif=01;35:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:*~=00;90:*#=00;90:*.bak=00;90:*.crdownload=00;90:*.dpkg-dist=00;90:*.dpkg-new=00;90:*.dpkg-old=00;90:*.dpkg-tmp=00;90:*.old=00;90:*.orig=00;90:*.part=00;90:*.rej=00;90:*.rpmnew=00;90:*.rpmorig=00;90:*.rpmsave=00;90:*.swp=00;90:*.tmp=00;90:*.ucf-dist=00;90:*.ucf-new=00;90:*.ucf-old=00;90:","XDG_CURRENT_DESKTOP":"X-Cinnamon","KONSOLE_DBUS_SERVICE":":1.104","HOMEBREW_NO_ENV_HINTS":"1","VSCODE_IPC_HOOK":"/run/user/1000/vscode-04b39de9-1.85-main.sock","CLOUDSDK_ROOT_DIR":"/opt/google-cloud-cli","KONSOLE_DBUS_SESSION":"/Sessions/9","VSCODE_CLI":"1","gopath":"/home/nowaker/projects/go","KONSOLE_VERSION":"230804","CHROME_DESKTOP":"code-url-handler.desktop","CLOUDSDK_PYTHON":"/usr/bin/python","rvm_bin_path":"/home/nowaker/.rvm/bin","GEM_PATH":"/home/nowaker/.rvm/gems/ruby-3.3.0:/home/nowaker/.rvm/gems/ruby-3.3.0@global","GEM_HOME":"/home/nowaker/.rvm/gems/ruby-3.3.0","XDG_SESSION_CLASS":"greeter","HOMEBREW_NO_ANALYTICS":"1","LC_IDENTIFICATION":"en_US.UTF-8","TERM":"xterm-256color","LESS_TERMCAP_mb":"\u001b[01;31m","LESS_TERMCAP_me":"\u001b[0m","LESS_TERMCAP_md":"\u001b[01;31m","LIBVIRT_DEFAULT_URI":"qemu:///system","GOOGLE_CLOUD_SDK_HOME":"/opt/google-cloud-cli","USER":"nowaker","SDL_AUDIODRIVER":"pulse","PYTHON":"/usr/bin/python","COLORFGBG":"15;0","DISPLAY":":0.0","VSCODE_PID":"161766","LESS_TERMCAP_ue":"\u001b[0m","SHLVL":"2","LESS_TERMCAP_us":"\u001b[01;32m","PAGER":"less","LC_TELEPHONE":"en_US.UTF-8","CVS_RSH":"ssh","LC_MEASUREMENT":"en_US.UTF-8","VSCODE_CWD":"/home/nowaker/projekty/modern/worldmodern","XDG_VTNR":"7","DESKTOP_AUTOSTART_ID":"10ae9e882d65f9acbb170594272665819400000016810008","XDG_SESSION_ID":"1","rvm_ruby_string":"ruby-3.3.0","MOZ_PLUGIN_PATH":"/usr/lib/mozilla/plugins","LC_CTYPE":"en_US.UTF-8","VSCODE_CRASH_REPORTER_PROCESS_TYPE":"extensionHost","XDG_RUNTIME_DIR":"/run/user/1000","IDN_DISABLE":"0","DEBUGINFOD_URLS":"https://debuginfod.archlinux.org ","LC_TIME":"en_US.UTF-8","ELECTRON_NO_ATTACH_CONSOLE":"1","XDG_DATA_DIRS":"/home/nowaker/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share","GDK_BACKEND":"x11","PATH":"/home/nowaker/.rvm/gems/ruby-3.3.0/bin:/home/nowaker/.rvm/gems/ruby-3.3.0@global/bin:/home/nowaker/.rvm/rubies/ruby-3.3.0/bin:/home/nowaker/.rvm/bin:/home/nowaker/projects/dreamhost/kubekick:/home/nowaker/.local/bin:/home/nowaker/bin:/home/nowaker/.local/bin:/opt/google-cloud-cli/bin:/bin:/usr/bin:/usr/local/bin:/usr/local/sbin:/usr/lib/jvm/default/bin:/opt/kde/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl","ORIGINAL_XDG_CURRENT_DESKTOP":"X-Cinnamon","DBUS_SESSION_BUS_ADDRESS":"unix:path=/run/user/1000/bus","VSCODE_NLS_CONFIG":"{\"locale\":\"en-us\",\"osLocale\":\"en-us\",\"availableLanguages\":{},\"_languagePackSupport\":true}","HG":"/usr/bin/hg","MAIL":"/var/mail/nowaker","IRBRC":"/home/nowaker/.rvm/rubies/ruby-3.3.0/.irbrc","GIO_LAUNCHED_DESKTOP_FILE_PID":"4086","GIO_LAUNCHED_DESKTOP_FILE":"/home/nowaker/.config/autostart/yakuake.desktop","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","rvm_path":"/home/nowaker/.rvm","LC_NUMERIC":"en_US.UTF-8","OLDPWD":"/home/nowaker/projekty/modern/worldmodern","GOPATH":"/home/nowaker/projects/go","TF_CLI_ARGS_backup":"-","BUNDLE_GEMFILE":"/home/nowaker/projekty/modern/worldmodern/.ruby-lsp/Gemfile"}
```

Executing the exact same command from the log `bundle exec rdbg --open --command --sock-path=/tmp/ruby-lsp-debug-sockets/ruby-debug-worldmodern-0.sock -- puma` succeeds. And I can attach a debugger from there with no issue, so something isn't adding up.

```json
    {
      "name": "Rails (attach)",
      "type": "ruby_lsp",
      "request": "attach",
    }
```

I tried bumping the dependencies, but it didn't help. Still exit 1.

```
nowaker@nwkr-desktop ~/projekty/modern/worldmodern (git)-[rails71] % bundle update debug
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Using debug 1.9.1 (was 1.8.0)
Bundle updated!
Gems in the group 'skip_on_arch_linux' were not updated.
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
nowaker@nwkr-desktop ~/projekty/modern/worldmodern (git)-[rails71] % bundle update ruby-lsp-rails
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Using prism 0.19.0 (was 0.18.0)
Fetching sorbet-runtime 0.5.11214 (was 0.5.11150)
Installing sorbet-runtime 0.5.11214 (was 0.5.11150)
Using ruby-lsp 0.13.4 (was 0.13.1)
Fetching ruby-lsp-rails 0.2.9 (was 0.2.8)
Installing ruby-lsp-rails 0.2.9 (was 0.2.8)
Bundle updated!
Gems in the group 'skip_on_arch_linux' were not updated.
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
nowaker@nwkr-desktop ~/projekty/modern/worldmodern (git)-[rails71] % bundle update ruby-lsp-rspec
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Fetching ruby-lsp-rspec 0.1.9 (was 0.1.8)
Installing ruby-lsp-rspec 0.1.9 (was 0.1.8)
Bundle updated!
Gems in the group 'skip_on_arch_linux' were not updated.
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
```

Neither Ruby 3.3.0 nor Ruby 3.1.4 works on this computer. Linux + RVM.

The exact same application with `launch.json` on my other computer works. Mac + RVM.

**Comment by vinistock** [comment]
We have shipped better RVM support in the stable version of the extension. If you can try this again, hopefully the debug console will print whatever errors occurred in the server, which can help us understand what's going on.

**Comment by Nowaker** [comment]
Aight, thank you @vinistock.

**Comment by GoldenSheep402** [comment]
I meet the same problem, even the .ruby-lsp folder is not created! And I will exit if I click the debug button

**Comment by Nowaker** [comment]
@vinistock It turned it this issue wasn't caused by RVM. I've not been encountering this issue for a while now, until I was able to trigger it again. Here's what happened.

First, I decided to bump `appsignal` gem:

```
modified: Gemfile.lock
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
-    appsignal (3.13.1)
+    appsignal (4.0.8)
```

`bundle exec puma` would normally use it. However, I noticed that Ruby LSP Debugger would load an older version of `appsignal`, inconsistent with what my `Gemfile.lock` wants!

So I set a breakpoint before `Appsignal.set_error`, then step into it, and stackframes point to `appsignal-3.13.1`. 

![Image](https://github.com/user-attachments/assets/10e044ea-d21c-4556-9c6d-6736cac6a0d4)

This shouldn't really ever happen, given Ruby LSP uses Bundler when present on the project. **But it is happening.** So I do `gem uninstall appsignal -v 3.13.1` and... I get this again:

![Image](https://github.com/user-attachments/assets/e6c18ebd-cff3-417a-b6e8-ac4d6992319a)

So, I install it back with `gem install appsignal -v 3.13.1` , and debug starts as normal again.

I'm in a pickle here. Unsure why/how Ruby LSP decides it should use an incorrect version of a dependency... And how it comes up with that version number to use in particular, given that gem version doesn't exist.

Any ideas what I could do, and what data to supply back, to help find the cause of this?

**Comment by Nowaker** [comment]
Whoa! **Ruby LSP does not obey Gemfile.lock!**

A change in `Gemfile.lock` was totally ignored:

```
-    appsignal (3.13.1)
+    appsignal (4.0.8)
```

It also wants a change in `Gemfile` to work:

```
-gem 'appsignal'
+gem 'appsignal', '~> 4.0'
```

So when `Gemfile` and `Gemfile.lock` are both modified, Ruby LSP picks up `appsignal-4.0.8` just fine - debugger starts, and I can step into it.

![Image](https://github.com/user-attachments/assets/084f5709-cc9c-46fd-b1f0-aec970fdda19)


...This is of course totally unexpected, since `Gemfile.lock` is in charge of deciding the exact versions of dependencies to use. :)

Now that I know the root cause, I can keep the version number of this dependency in my Gemfile, but let's take a step back and come up not just with a fix, but a generic logging mechanism that could help catch problems like these. What could "Ruby LSP" print in "Output" window to help understand the issue at hand? What's currently printed there is irrelevant to debugging this. And that exit 1 comes from somewhere... Nothing in stdout/stderr as Ruby LSP tries to run it? There could be a clue about missing `appsignal-3.13.1`. At least that would give a hint it's about dependency resolution.

**Comment by Nowaker** [comment]
I also noticed that after the first successful run of Ruby LSP with `appsignal-4.0.8`, I can now remove my `Gemfile` change and Ruby LSP is still activating `appsignal-4.0.8`. It no longer forces `appsignal-3.13.1` in. That sounds like a bug in Ruby LSP's caching mechanism, I presume.

**Comment by Nowaker** [comment]
Thanks for getting back to me @vinistock. Appreciate your help here.

> If your lockfile changes, we [restart](https://github.com/Shopify/ruby-lsp/blob/6fe556cb7529e8b840e484ad3f953528aa5379be/vscode/src/workspace.ts#L302) the language server, which re-runs the custom bundle logic, detects that the main lockfile has been modified and re-generates everything from scratch

Uhm... And if I execute `bundle update appsignal` when my VS Code / Ruby LSP isn't running, is my debugging broken forever now? ;-) Just like it was for me for a couple months before it randomly started working again on this computer.

> The puzzling part for me is: are you not seeing the language server restart automatically upon modifying the lockfile?

Unsure, since I use my own terminal to execute `bundle update <dependency name>` whenever needed, so my VS Code isn't in focus. I don't use a teeny tiny VS Code terminal on a daily basis.

> If your lockfile changes, we [restart](https://github.com/Shopify/ruby-lsp/blob/6fe556cb7529e8b840e484ad3f953528aa5379be/vscode/src/workspace.ts#L302) the language server, which re-runs the custom bundle logic

I use VS Code on a daily basis. I also turn off my computer every day. That means Ruby LSP starts from scratch many times. However, the issue persisted across Ruby LSP restarts. That indicates it's not really about `Gemfile.lock` being watched for changes, but rather, something in Ruby LSP activation code not picking up on the fact `Gemfile.lock` was modified, and using its stale version (or using some files/caches that were created based on the stale version of `Gemfile.lock`).

> I also just realized this: the Ruby LSP debug client only attaches the debugger to the running process, but it doesn't control which dependencies are loaded by that process. Are you restarting the process you want to attach the debugger to after upgrading the gem?

Please don't get distracted by `request=attach` being mentioned here. :-) It was to give you an example that I am able to run the debug fine if I attach to it, but `request=launch` gets me exit 1 and no indication about what is going on and why.

**This issue is solely about `request=launch` returning with exit status 1 when some `Gemfile.lock` changes were made that Ruby LSP isn't aware of.**

**Comment by vinistock** [comment]
> And if I execute bundle update appsignal or git pull when my VS Code / Ruby LSP isn't running, is my debugging broken forever now?
> ...
> Unsure, since I use my own terminal to execute bundle update <dependency name> whenever needed, so my > VS Code isn't in focus. Or I may be starting my work, and therefore git pull first, and then code .
> after that. I don't use a teeny tiny VS Code terminal on a daily basis.
> ...
> I use VS Code on a daily basis. I also turn off my computer every day. That means Ruby LSP starts from scratch many times

Whenever the server launches, be it due to a restart, re-opening VS Code, automatic file watcher restart or any other trigger, we always check if the contents of the main lockfile are not matching the contents of the custom lockfile, so that we can then re-copy and start from scratch ([see this line](https://github.com/Shopify/ruby-lsp/blob/a4f5a2d780950816829bd1372f559979d947e7cd/lib/ruby_lsp/setup_bundler.rb#L94)).

Updating a gem, even if you do it in a separate terminal or with the editor closed, should make no difference. When you open VS Code, it will compare the SHA digest of the previous lockfile content vs the current one and re-generate if needed.


Based on this comment

> It also wants a change in Gemfile to work:

I suspect there's some Bundler related issue at play. We don't even watch the Gemfile or do anything with it, since what controls the versions is the lockfile. The only impact that adding a constraint to the Gemfile would have is to force Bundler to resolve versions in a certain way.

Do you happen to have any global (`~/.bundle/config`) or project (`~/path/to/project/.bundle/config`) Bundler settings? We did just merge #2535, which fixes handling of Bundler setttings.

**Comment by Nowaker** [comment]
Wait, the plugin generates its own `Gemfile`?! So I've been right all along!

> using its stale version for its own demise instead (or using some files/caches that were created based on the stale version of Gemfile.lock)

Too bad I never saw `.ruby-lsp/Gemfile`! Given that, I'd like to re-raise the question I asked in my previous comment:

**Why [is this entire bundle setup thing happening for `request=launch`]? Since it's just an `rdbg` call, which requires `debug` in `Gemfile` (which mine has), there is absolutely no need for Ruby LSP to perform any of what I'm seeing in `RubyLsp::SetupBundler`. I understand it's important for the language server, especially to [keep] gem its version up to date [without requiring a constant `bundle update ruby-lsp-rails`], but I don't see how it has any use for `request=launch` in particular. [Especially now, knowing it causes unexplained exit 1 in certain situations.**

**Comment by malcolmohare** [comment]
There was a time when it didn't do this custom bundle for debugging, but then one day this: https://github.com/Shopify/ruby-lsp/commit/f55b903e603f9fdb6f34b54a28ea5578d01544de

It's a nice feature **IF** the users repo didn't include the debug gem, but when it does, its causing these issues.  I guess the history/usage of the custom bundle expanded over time so what was once an edge case is the every case.

**Comment by jklimke** [comment]
I noticed that the .ruby-lsp/Gemfile.lock was heavily outdated compared to my projects gemfile. 

As a quickfix to be able to debug deletion of this .ruby-lsp/Gemfile.lock or the whole .ruby-lsp folder worked for me to be able to debug again.

**Comment by malcolmohare** [comment]
Removing the chunk of code that sets `BUNDLE_GEMFILE` causes the debugging command to not work.  Changing that ENV to point at the workspace Gemfile instead of the ruby-lsp Gemfile did what I wanted.  I just need to figure out how to make that a per workspace setting or how to detect the presence of the debug gem in the main Gemfile. 🤔

**Comment by malcolmohare** [comment]
I've got a pull request up for adding a new setting, which allows users to specify the path to the Gemfile to use when running in debug.  If the setting is blank, nothing changes.

https://github.com/Shopify/ruby-lsp/pull/2707

This fixes the issue I was having, where the Gemfile the LSP generated in the lsp-ruby file had newer and incompatible versions of gems I used, and thus failed when parsing the code it was running.  This fixes that, and it works fine, as long as your Gemfile contains the debug gem.

**Comment by vinistock** [comment]
Just wanted to share an update: I managed to reproduce this with tests and I believe I have a fix for it. It's related to how the composed bundle environment is not reported back to the extension, which makes us try to launch the debugger with the wrong Bundler settings.

I'm waiting on PR reviews until I can submit the related changes. Once everything is in replace, I'll ask the folks here to test it out.

**Comment by vinistock** [comment]
Okay, I believe with [v0.8.14](https://github.com/Shopify/ruby-lsp/releases/tag/vscode-ruby-lsp-v0.8.14) for the extension and [v0.22.0](https://github.com/Shopify/ruby-lsp/releases/tag/v0.22.0) for the server, this problem will be fixed.

If that's not the case, please let us know.

**Comment by RolandMunsil** [comment]
I'm running into what seems to be the exact same issue on the latest versions of the extension and the server.

OS: Windows
Project has no bundle (though I tried adding a bundle and it didn't seem to fix anything)

![Image](https://github.com/user-attachments/assets/77de1f16-2360-4467-a553-0b4c489f6999)


launch.json:
```
{
    "type": "ruby_lsp",
    "name": "Ruby",
    "request": "launch",
    "program": "ruby ${file}",
}
```

Output when I try to launch:
```
2024-12-04 19:38:49.456 [info] Spawning debugger in directory e:\OneDrive\Code\AdventOfCode
2024-12-04 19:38:49.456 [info]    Command bundle exec rdbg --port 2964 --open --command -- ruby E:\OneDrive\Code\AdventOfCode\2024\01\d01p1.rb
2024-12-04 19:38:49.456 [info]    Environment { (snipped) }
```

And then I get the "Debugger exited with status 1" message.

**Comment by andyw8** [comment]
@RolandMunsil please log a [new issue](https://github.com/Shopify/ruby-lsp/issues/new?template=bug_template_vscode.yml) for this so that we can more easily track.

**Comment by RolandMunsil** [comment]
Actually, I seem to have ended up getting it working? I'm not sure exactly which things I tried were necessary, but removing `, platforms: :mri` from the Gemfile in `.ruby-lsp` seems to be important at least, because adding it back breaks things again. Should I still make an issue?

**Comment by vinistock** [comment]
@RolandMunsil please do create the new issue. I thought `platform: :mri` would include Windows. After all, the Ruby you're using on Windows is also MRI (CRuby). The only reason we scope the platform is because the `debug` gem is not compatible with JRuby or TruffleRuby.

Maybe there's some RubyGems gotcha with that platform value and we need to broaden it to include Windows too.
